tests/integration/security/citadel/namespace_targeting_test.go

```diff
@@ -0,0 +1,86 @@
+package citadel
+
+import (
+	"fmt"
+	"istio.io/istio/pkg/test"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	v1 "k8s.io/api/core/v1"
+	mv1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/citadel"
+)
+
+const (
+	testNsPrefix       = "targeting-test"
+	serviceAccountName = "sample-service-account"
+)
+
+// TestNamespaceTargeting verifies that Citadel respects the designated namespace targeting labels.
+// current implementation is closely tied with Kubernetes concepts, but this is difficult to avoid
+// until Citadel itself is less closely tied with Kubernetes concepts
+func TestNamespaceTargeting(t *testing.T) {
+	framework.NewTest(t).Run(func(ctx framework.TestContext) {
+
+		citadel.NewOrFail(t, ctx, citadel.Config{Istio: ist})
+		env := ctx.Environment().(*kube.Environment)
+
+		// Label combinations
+		type citadelLabels map[string]string
+		testCases := map[string]struct {
+			labels   citadelLabels
+			targeted bool
+		}{
+			"override label set to false, env label matches => should not generate secret": {
+				labels: citadelLabels{
+					"ca.istio.io/override": "false",
+				},
+				targeted: false,
+			},
+			"no override label present, env label matches => should generate secret": {
+				labels: citadelLabels{
+					"ca.istio.io/env": "istio-system", // how can we get this from config?
```

Comment:
Are you asking how to list information about the namespaces in a test, like kubectl?

Reply:
`ca.isito.io/env` label is meant to target the namespace the desired Citadel instance is running in, so I'm wondering if there's a way to find out, in this integration test, where the control plane is running, or if its safe to assume `istio-system`?

Reply:
Does ist.Settings().SystemNamespace help?

Reply:
That's exactly what I was looking for, thanks :+1: